data/types/blood/glucose/glucose.go

```diff
@@ -33,3 +36,75 @@ func (g *Glucose) Normalize(normalizer data.Normalizer) {
 		g.Value = dataBloodGlucose.NormalizeValueForUnits(g.Value, units)
 	}
 }
+
+func roundToEvenWithDecimalPlaces(v float64, decimals int) float64 {
+	if decimals < 1 {
+		return math.RoundToEven(v)
+	}
+	coef := math.Pow(10, float64(decimals))
+	return math.RoundToEven(v*coef) / coef
+}
+
+type Classification string
+
+const (
+	ClassificationInvalid Classification = "invalid"
+
+	VeryLow       = "very low"
+	Low           = "low"
+	OnTarget      = "on target"
+	High          = "high"
+	VeryHigh      = "very high"
+	ExtremelyHigh = "extremely high"
+)
+
+type classificationThreshold struct {
+	Name      Classification
+	Value     float64
+	Inclusive bool
+}
+
+type Classifier []classificationThreshold
+
+func (c Classifier) Classify(g *Glucose) (Classification, error) {
+	normalized, err := dataBloodGlucose.NormalizeValueForUnitsSafer(g.Value, g.Units)
+	if err != nil {
+		return ClassificationInvalid, errors.Wrap(err, "unable to classify")
+	}
+	// Rounded values are used for all classifications. To not do so risks introducing
+	// inconsistency between frontend, backend, and/or other reports. See BACK-3800 for
+	// details.
+	rounded := roundToEvenWithDecimalPlaces(normalized, 1)
+	for _, threshold := range c {
+		if threshold.Includes(rounded) {
+			return threshold.Name, nil
+		}
+	}
+	// Ensure your highest threshold has a value like math.MaxFloat64 to avoid this.
+	return ClassificationInvalid, errors.Newf("unable to classify value: %v", *g)
+}
+
+// TidepoolADAClassificationThresholdsMmolL for classifying glucose values.
+//
+// All values are normalized to MmolL before classification.
+//
+// In addition to the standard ADA ranges, the Tidepool-specifiic "extremely high" range is
+// added.
+//
+// It is the author's responsibility to ensure the thresholds remain sorted from smallest to
+// largest.
+var TidepoolADAClassificationThresholdsMmolL = Classifier([]classificationThreshold{
+	{Name: VeryLow, Value: 3, Inclusive: false},                    // Source: https://doi.org/10.2337/dc24-S006
+	{Name: Low, Value: 3.9, Inclusive: false},                      // Source: https://doi.org/10.2337/dc24-S006
+	{Name: OnTarget, Value: 10, Inclusive: true},                   // Source: https://doi.org/10.2337/dc24-S006
```

Comment:
Will we ever have different values for `Inclusive` for a specific classifier or will the classification itself determine the inclusiveness. That is, is `High` always inclusive, or can I have a classifier where it is not?

Reply:
> Will we ever have different values for `Inclusive` for a specific classifier or will the classification itself determine the inclusiveness. That is, is `High` always inclusive, or can I have a classifier where it is not?

I don't know. I know there are "custom" ranges coming down the pipe, which is why I've kept the possibility.